Re-enable CheckInternalHash in htmltest and fail on new errors

DamianMaslanka5 opened the pull request:
These changes enable CheckInternalHash again, now the check fails for new errors to avoid introducing new broken links, existing errors are ignored

TODO remove test commit a8b79f6fec909babde25c7520042dfb13472bebe before merging

Related to 
* https://github.com/ClickHouse/ClickHouse/pull/65248 initial PR with htmltest
* https://github.com/ClickHouse/clickhouse-docs/issues/2398 issue for broken links 
* https://github.com/ClickHouse/ClickHouse/pull/72743 PR which disables CheckInternalHash

cc: @Felixoid 

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)



> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ] <!---ci_set_required--> Allow: All Required Checks
- [ ] <!---ci_include_stateless--> Allow: Stateless tests
- [ ] <!---ci_include_stateful--> Allow: Stateful tests
- [ ] <!---ci_include_integration--> Allow: Integration Tests
- [ ] <!---ci_include_performance--> Allow: Performance tests
- [ ] <!---ci_set_builds--> Allow: All Builds
- [ ] <!---batch_0_1--> Allow: batch 1, 2 for multi-batch jobs
- [ ] <!---batch_2_3--> Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ] <!---ci_exclude_style--> Exclude: Style check
- [ ] <!---ci_exclude_fast--> Exclude: Fast test
- [ ] <!---ci_exclude_asan--> Exclude: All with ASAN
- [ ] <!---ci_exclude_tsan|msan|ubsan|coverage--> Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ] <!---ci_exclude_aarch64|release|debug--> Exclude: All with aarch64, release, debug
---
- [ ] <!---ci_include_fuzzer--> Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ] <!---ci_exclude_ast--> Exclude: AST fuzzers
---
- [ ] <!---do_not_test--> Do not test
- [ ] <!---woolen_wolfdog--> Woolen Wolfdog
- [ ] <!---upload_all--> Upload binaries for special builds
- [ ] <!---no_merge_commit--> Disable merge-commit
- [ ] <!---no_ci_cache--> Disable CI cache

DamianMaslanka5 commented on tests/ci/docs_check.py:
```diff
@@ -107,7 +107,7 @@ def main():
                 if "ERROR" in line:
                     build_status = FAIL
-                    job_status = FAIL
+                    job_status = FAILURE
```
This assert from `tests/ci/report.py` fails when `FAIL` is used

```
    def __post_init__(self):
        assert self.status in (SUCCESS, ERROR, FAILURE, PENDING)
```

robot-ch-test-poll commented:
<!-- automatic status comment for PR #73705 from DamianMaslanka5/ClickHouse:html-test-ignore-existing -->
*This is an automated comment for commit a8b79f6fec909babde25c7520042dfb13472bebe with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73705/a8b79f6fec909babde25c7520042dfb13472bebe/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/73705/a8b79f6fec909babde25c7520042dfb13472bebe/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

DamianMaslanka5 commented:
@Felixoid in 7e09974eae250f7104ec804953c0d7895880e26a I fixed errors reported by pylint and black

DamianMaslanka5 commented:
@alexey-milovidov I see that you run the CI checks

The results on github look correctly, and I think this can be merged.

The only problem with current code is that the new errors are not reported in the [test reports](https://s3.amazonaws.com/clickhouse-test-reports/73705/7e09974eae250f7104ec804953c0d7895880e26a/docs_check//home/ubuntu/actions-runner/_work/_temp/test/docs_check/htmltest.log), if you think this is important I can improve the output in this PR or open another PR.

I think it would be the best to merge this ASAP to avoid introducing more issues.

Let me know what you want me to do.

If you want to merge current code you need to:

1. Rebase this branch onto master, remove a8b79f6fec909babde25c7520042dfb13472bebe (or just uncomment the line and squash the commits).
1. Make sure there are no other "new" errors (rebuild the docs and run the script locally).
1. Add all new errors to `htmltest_existing_errors`

I wanted to see these errors in the output to confirm that the script works correctly, see a8b79f6fec909babde25c7520042dfb13472bebe
```
INFO:root:new error: hash does not exist --- ru/operations/settings/permissions-for-queries.html --> #permissions_for_queries
INFO:root:new error: hash does not exist --- ru/operations/settings/permissions-for-queries.html --> #permissions_for_queries
```

There are some new errors which were introduced after opening this PR

```
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max_concurrent_queries_for_all_users
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max_concurrent_queries_for_user
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max_concurrent_queries_for_all_users
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max-table-size-to-drop
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max_temporary_data_on_disk_for_user
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #max_temporary_data_on_disk_size_for_query
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #server-settings_zookeeper
INFO:root:new error: hash does not exist --- en/operations/server-configuration-parameters/settings.html --> #total-memory-profiler-step
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-1
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-2
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-3
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-1
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-2
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-3
INFO:root:new error: hash does not exist --- en/integrations/javascript.html --> #fn-2
```

DamianMaslanka5 committed:
Re-enable CheckInternalHash in htmltest and ignore existing errors

Replace FLAKY with FAIL and check only for new errors

DamianMaslanka5 commented:
@alexey-milovidov I rebased the branch, squashed the commits, added new errors from https://github.com/ClickHouse/ClickHouse/pull/73705#issuecomment-2568041189 to `htmltest_existing_errors`, so this should be now ready to merge without any additional changes to `htmltest_existing_errors`

There was one more new error `"hash does not exist --- en/sql-reference/statements/create/view.html --> #window-view-experimental",` introduced in https://github.com/ClickHouse/ClickHouse/pull/73978

cc: @gingerwizard
